presto-main/src/main/java/com/facebook/presto/operator/MultiChannelGroupByHash.java

```diff
@@ -137,6 +141,8 @@ public MultiChannelGroupByHash(
 
         groupAddressByGroupId = new LongBigArray();
         groupAddressByGroupId.ensureCapacity(maxFill);
+
+        this.updateReservedMemory = requireNonNull(updateReservedMemory, "updateReservedMemory is null");
```

Comment:
why don't we reserve memory for the data structures that we allocate in this constructor (`hashCapacity`-sized stuff)?

Reply:
ditto